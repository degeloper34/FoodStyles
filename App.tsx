import {ApolloClient, ApolloProvider, InMemoryCache} from "@apollo/client";
import {StatusBar} from "expo-status-bar";
import {SafeAreaProvider} from "react-native-safe-area-context";
import useCachedResources from "./src/hooks/useCachedResources";
import Navigation from "./src/navigation";

const token =
  "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6NTI3LCJ0eXBlIjoiQUNDRVNTIiwiaWF0IjoxNjU3MTQ0MzAyLCJleHAiOjE2NTc3NDkxMDJ9.XxvR4ah2U6NeUFAxKtI-sSzByFfbHGcz06LGpp1FMPVCar6mQyfctQgtKOkg2w5Wv_luULa0SgcSovS6VNUB9_BBlFdH7PYBjQO3N1N0Dwi53FBzA0rdnSkRGdj5Oe7hQ-9fshljlo5cJUYBbz4N0Ndpt323V5Eu8lGqJZ8Z5husoCiK93fZ0l3-Vr0M_sLvulZAELXGRG1QmYCwMwrp43HF9zyOLjcSHPVCvbpettqGxGv8Gne7O-XbykNv8ehG81vPcJDMSImwXGY8Yd47EoIrxdoN-3IlJumBK1Ql29hy5N7HBbk2oil3KMwBbw4MowB7rdlaT1i5_5Xr6q0iIg";

export const client = new ApolloClient({
  uri: "https://api-dev.foodstyles.com/graphql",
  cache: new InMemoryCache(),
  headers: {
    authorization: `Bearer ${token}`,
  },
});

export default function App() {
  const isLoadingComplete = useCachedResources();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      <ApolloProvider client={client}>
        <SafeAreaProvider>
          <StatusBar style={"light"} />
          <Navigation />
        </SafeAreaProvider>
      </ApolloProvider>
    );
  }
}
