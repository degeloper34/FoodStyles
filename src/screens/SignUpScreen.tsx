import {useMutation} from "@apollo/client";
import {LinearGradient} from "expo-linear-gradient";
import {Pressable, StyleSheet} from "react-native";
import {CustomText, Loading} from "../components/atoms";
import {Spacing, TextStyles} from "../styles";
import {useHeaderHeight} from "@react-navigation/elements";
import {SIGN_UP_WITH_EMAIL_MUTATION} from "../api/mutations";
import {Controller, useForm} from "react-hook-form";
import {CustomTextInput} from "../components/molecules";
import {SignUpForm} from "../../types";

export default function SignUpScreen() {
  const headerHeight = useHeaderHeight();

  const {handleSubmit, control, formState} = useForm<SignUpForm>({
    mode: "onChange",
    defaultValues: {
      name: "",
      email: "",
      password: "",
    },
  });

  const [signUpWithEmail, {loading}] = useMutation(SIGN_UP_WITH_EMAIL_MUTATION);

  const onPressSignUp = async (form: SignUpForm) => {
    await signUpWithEmail({
      variables: {
        name: form?.name,
        email: form?.email,
        password: form?.password,
      },
    });
  };

  const {container, btnSignUp} = styles;

  return (
    <LinearGradient
      colors={["rgb(250,119,69)", "rgb(243, 196, 66)"]}
      style={{paddingTop: headerHeight + 12, ...container}}
    >
      {loading && <Loading />}

      <Controller
        control={control}
        name="name"
        rules={{
          required: true,
        }}
        render={({field: {onChange, value}}) => (
          <CustomTextInput
            text="Your name"
            value={value}
            onChangeText={onChange}
            maxLength={50}
          />
        )}
      />

      <Controller
        control={control}
        name="email"
        rules={{
          required: true,
          pattern: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
        }}
        render={({field: {onChange, value}}) => (
          <CustomTextInput
            text="Email"
            value={value}
            onChangeText={onChange}
            maxLength={50}
          />
        )}
      />

      <Controller
        control={control}
        name="password"
        rules={{
          required: true,
          minLength: 6,
        }}
        render={({field: {onChange, value}}) => (
          <CustomTextInput
            text="Password (min 6 characters)"
            value={value}
            onChangeText={onChange}
            maxLength={50}
            secureTextEntry
          />
        )}
      />

      <Pressable
        style={btnSignUp}
        onPress={handleSubmit(onPressSignUp)}
        disabled={!formState.isValid}
      >
        <CustomText
          text={"SIGN UP"}
          type={"bold"}
          style={TextStyles.TEXT_STYLE}
        />
      </Pressable>
    </LinearGradient>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: Spacing.xxxl,
  },
  btnSignUp: {
    backgroundColor: "rgb(17,206,144)",
    alignItems: "center",
    borderRadius: 100,
    paddingVertical: Spacing.s,
    marginHorizontal: 80,
  },
});
