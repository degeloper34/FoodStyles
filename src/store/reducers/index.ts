import {combineReducers} from "redux";

import memberReducers from "./memberReducers";

export const rootReducer = combineReducers({
  memberReducers,
});
