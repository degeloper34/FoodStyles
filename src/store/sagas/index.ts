import {all} from "redux-saga/effects";

import {memberWatcher} from "./memberSagas";

export default function* rootSaga() {
  yield all([memberWatcher()]);
}
