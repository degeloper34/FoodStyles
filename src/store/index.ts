export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

import {legacy_createStore as createStore} from "redux";
import {rootReducer} from "./reducers";

export const store = createStore(rootReducer);
