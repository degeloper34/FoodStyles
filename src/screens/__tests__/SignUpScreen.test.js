import renderer from "react-test-renderer";
import React from "react";
import {Provider} from "react-redux";
import {store} from "../../store";
import {ApolloProvider} from "@apollo/client";
import {apolloClient} from "../../../App";
import SignUpScreen from "../SignUpScreen";

describe("SignUpScreen Tests", () => {
  test("SignUpScreen | Render without issues", () => {
    const screen = renderer
      .create(
        <ApolloProvider client={apolloClient}>
          <Provider store={store}>
            <SignUpScreen />
          </Provider>
        </ApolloProvider>
      )
      .toJSON();

    expect(screen).toMatchSnapshot();
  });
});
