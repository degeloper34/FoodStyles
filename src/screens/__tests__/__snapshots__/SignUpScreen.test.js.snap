// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpScreen Tests SignUpScreen | Render without issues 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4294604613,
        4294165570,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 32,
      "paddingTop": NaN,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {
          "fontFamily": "proxima-semibold",
          "textDecorationLine": "none",
        },
      ]
    }
  >
    Your name
  </Text>
  <TextInput
    maxLength={50}
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "fontFamily": "proxima-regular",
        "marginBottom": 16,
        "marginTop": 8,
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      }
    }
    value=""
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {
          "fontFamily": "proxima-semibold",
          "textDecorationLine": "none",
        },
      ]
    }
  >
    Email
  </Text>
  <TextInput
    maxLength={50}
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "fontFamily": "proxima-regular",
        "marginBottom": 16,
        "marginTop": 8,
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      }
    }
    value=""
  />
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "600",
          "letterSpacing": 0,
        },
        Object {
          "fontFamily": "proxima-semibold",
          "textDecorationLine": "none",
        },
      ]
    }
  >
    Password (min 6 characters)
  </Text>
  <TextInput
    maxLength={50}
    onChangeText={[Function]}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": 4,
        "fontFamily": "proxima-regular",
        "marginBottom": 16,
        "marginTop": 8,
        "paddingHorizontal": 8,
        "paddingVertical": 12,
      }
    }
    value=""
  />
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(17,206,144)",
        "borderRadius": 100,
        "marginHorizontal": 80,
        "paddingVertical": 12,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontSize": 16,
            "fontStyle": "normal",
            "fontWeight": "600",
            "letterSpacing": 0,
          },
          Object {
            "fontFamily": "proxima-bold",
            "textDecorationLine": "none",
          },
        ]
      }
    >
      SIGN UP
    </Text>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
